src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -16,16 +16,27 @@ def target_project_base_name
   end
 
   def branch_package(workflow_filters = {})
-    create_branched_package if scm_webhook.new_pull_request? || (scm_webhook.updated_pull_request? && target_package.blank?) || scm_webhook.push_event? || scm_webhook.tag_push_event?
-
-    add_branch_request_file(package: target_package)
-
+    scmsync = false
+    
+    # if the whole project is scm synced we find the `Project` object but no `Package` objects (they only exist in the backend)
+    # if the scmsync is setup on the package level, the `Package` object exists
+    source_project = Project.get_by_name(source_project_name) 
+    source_package = Package.get_by_project_and_name(source_project_name, source_package_name)
+
+    scmsync = true if source_project.try(:scmsync).present? || source_package.try(:scmsync).present?
+    create_branched_package(scmsync: scmsync) if scm_webhook.new_pull_request? || (scm_webhook.updated_pull_request? && target_package.blank?) || scm_webhook.push_event? || scm_webhook.tag_push_event?
+    add_branch_request_file(package: target_package) unless scmsync
+    
+    add_scmsync_to_meta(package: target_package, scmsync_url: source_package.scmsync) if source_package.try(:scmsync).present?
```

Comment:
meta? That is just the name of the Tab in the UI...

Reply:
yes, that was just the first name I had in mind, its changed now